packages/appium/lib/cli/parser.js

```diff
@@ -280,6 +287,38 @@ class ArgParser {
       }
     }
   }
+
+  /**
+   * Add subcommand and sub-sub commands for 'setup' subcommand.
+   * @param {import('argparse').SubParser} subParser
+   */
+    static _addSetupToParser(subParser) {
+      const setupParser = subParser.add_parser('setup', {
+        add_help: true,
+        help: `Install latest drivers '${_.join(DEFAULT_DRIVERS, ',')}' and ` +
+          `plugins '${_.join(DEFAULT_PLUGINS, ',')}' ` +
+          `if APPIUM_HOME has no drivers and plugins`,
+      });
+
+      ArgParser._patchExit(setupParser);
+      const extSubParsers = setupParser.add_subparsers({
+        dest: `setupCommand`,
+      });
+
+      const parserSpecs = [
+        {
+          command: 'all',
```

Comment:
I am not quite convinced by having this all/not alll parameter.
Maybe we could do it more obvious by having presets, for example

setup mobile -> installs uia2 (and xcuitest if macOS) (default)
setup desktop -> installs mac2/windows (depending on current platform)
setup mobile+plugins -> installs uia2 and xcuitest with officially supported plugins
setup browsers -> chromium, gecko (and safari if macOS)
etc...

Reply:
Sounds good. I like the mobile, desktop etc...

Let me write `mobile`, `desktop` and `browser` as an example subcommands